packages/webview_flutter/lib/src/tizen_webview.dart

```diff
@@ -0,0 +1,228 @@
+// Copyright 2023 Samsung Electronics Co., Ltd. All rights reserved.
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:flutter_tizen/widgets.dart';
+import 'package:webview_flutter_platform_interface/webview_flutter_platform_interface.dart';
+
+/// Signature for callbacks that report a method calls by navigation delegate method channel.
+typedef NavigationDelegateMethodChannelCallback = Future<dynamic> Function(
+    MethodCall method);
+
+/// A Tizen webview that displays web pages.
+class TizenWebview {
+  /// Constructs a new WebView.
+  TizenWebview()
+      : _controllerChannel =
+            const MethodChannel('plugins.flutter.io/tizen_webview_controller'),
+        _navigationDelegateChannel = const MethodChannel(
+            'plugins.flutter.io/tizen_webview_navigation_delegate') {
+    _controllerChannel.setMethodCallHandler(_onMethodCall);
+  }
+
+  final MethodChannel _controllerChannel;
+  final MethodChannel _navigationDelegateChannel;
+  bool _isCreated = false;
+  bool _hasNavigationDelegate = false;
+  final Map<String, JavaScriptChannelParams> _javaScriptChannelParams =
+      <String, JavaScriptChannelParams>{};
+  final Map<String, dynamic> _pendingMethodCalls = <String, dynamic>{};
+
+  Future<bool?> _onMethodCall(MethodCall call) async {
+    switch (call.method) {
+      case 'javaScriptChannelMessage':
+        final String channel = call.arguments['channel']! as String;
+        final String message = call.arguments['message']! as String;
+        if (_javaScriptChannelParams.containsKey(channel)) {
+          _javaScriptChannelParams[channel]
+              ?.onMessageReceived(JavaScriptMessage(message: message));
+        }
+
+        return true;
+    }
+
+    throw MissingPluginException(
+      '${call.method} was invoked but has no handler',
+    );
+  }
+
+  Future<T?> _invokeChannelMethod<T>(String method, [dynamic arguments]) async {
+    if (!_isCreated) {
+      _pendingMethodCalls[method] = arguments;
+      return Future<T?>(() => null);
```

Comment:
Maybe just
```suggestion
      return null;
```

Reply:
done thank you

Reply:
Just be aware that adding a method call that is expected to return a value to a pending list is actually a bit pointless (since the method will be executed later and the caller will not have an access to the return value).